File: 151003/Shafarenko/lab3/cassandra/Publisher/Entity/Db/Post.cs
using Publisher.Entity.Common;
using System.ComponentModel.DataAnnotations;

namespace Publisher.Entity.Db
{
    public class Post(string content) : AbstractEntity
    {
        public int NewsId { get; set; }
        [MinLength(2)]
        public string Content { get; set; } = content;
        public News News { get; set; } = null!;

        public Post() : this(string.Empty) { }
        public Post(int id, int newsId, string content, News news) : this(content)
        {
            Id = id;
            NewsId = newsId;
            News = news;
        }
    }
}

File: 151002/Luschickij/lab1/DC_lab1/src/main/java/com/luschickij/DC_lab/model/entity/Post.java
package com.luschickij.DC_lab.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "tbl_post")
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JoinColumn(name = "news_id")
    private News news;

    @Column(name = "name", length = 32)
    private String content;
}

